Source/UI/Search/SearchResultsView.swift

```diff
@@ -0,0 +1,239 @@
+                                        switch identityOrAbout {
+                                        case .left(let identity):
+                                            IdentityButton(identity: identity, style: .golden)
+                                        case .right(let about):
+                                            IdentityButton(identity: about.identity, style: .golden)
+                                        }
+                                    }
+                                }
+                            }
+                        case .people:
+                            SearchResultsGrid {
+                                ForEach(searchResults.users) { identityOrAbout in
+                                    switch identityOrAbout {
+                                    case .left(let identity):
+                                        IdentityButton(identity: identity, style: .golden)
+                                    case .right(let about):
+                                        IdentityButton(identity: about.identity, style: .golden)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    Spacer(minLength: 0)
+                }
+            }
+        }
+        .task(id: searchText) {
+            guard searchResults.data.isReadyToSearch || searchResults.query != searchText else {
+                return
+            }
+            searchResults = SearchResults(data: .loading, query: searchText)
+            searchResults = await fetchSearchResults(for: searchText)
+            selectedSection = searchResults.activeSections.first ?? .all
+        }
+    }
+
+    // MARK: - Loading Search Results
+
+    private func fetchSearchResults(for query: String) async -> SearchResults {
+        guard !query.isEmpty else {
+            return SearchResults(data: .none, query: query)
+        }
+
+        let normalizedQuery = query.lowercased().trimmingCharacters(in: .whitespacesAndNewlines)
+        let identifier: Identifier = query
+        let isFeedIdentifier = identifier.isValidIdentifier && identifier.sigil == .feed
+        let isMessageIdentifier = identifier.isValidIdentifier && identifier.sigil == .message
+
+        if isFeedIdentifier {
+            return SearchResults(data: .feedID(identifier), query: query)
+        } else if isMessageIdentifier {
+            return SearchResults(data: .messageID(await loadMessage(with: identifier)), query: query)
+        } else {
+            async let abouts = loadAbouts(matching: normalizedQuery)
+            async let posts = loadPosts(matching: normalizedQuery)
+            return SearchResults(data: .universal(people: await abouts, posts: await posts), query: query)
+        }
+    }
+
+    func loadAbouts(matching filter: String) async -> [About] {
+        do {
+            return try await Bots.current.abouts(matching: filter)
```

Comment:
Can we get the Bot from the environment rather than using the singleton here and below?

Reply:
Good catch.